### PR TITLE
Create policy for qatlib

### PR DESCRIPTION
**Pull request by 5umm3r15**
Intel QuickAssist Technology Library (QATlib) contains user space libraries that allow access for Intel QAT devices.

Resolves: rhbz#2080443

### CODE REVIEW

**Comment by 5umm3r15** [comment]
Copr build:
dnf copr enable nknazeko/qatlib-selinux

**Comment by 5umm3r15** [comment]
@zpytela can you please look at this?

File: policy/modules/contrib/qatlib.fc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/usr/sbin/qat_init.sh			--	gen_context(system_u:object_r:qatlib_exec_t,s0)
```
Comment (zpytela) [inline policy/modules/contrib/qatlib.fc]: Dot should be escaped.

File: policy/modules/contrib/qatlib.fc [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+/usr/sbin/qatmgr			--	gen_context(system_u:object_r:qatlib_exec_t,s0)
+
+/usr/lib/systemd/system/qat\.service	--	gen_context(system_u:object_r:qatlib_unit_file_t,s0)
+
+/run/qat/qatmgr\.pid			--	gen_context(system_u:object_r:qatlib_var_run_t,s0)
```
Comment (zpytela) [inline policy/modules/contrib/qatlib.fc]: /var/run still needs to be used

File: policy/modules/contrib/qatlib.te [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#
+# Declarations
+#
+
+type qatlib_t;
+type qatlib_exec_t;
+init_daemon_domain(qatlib_t, qatlib_exec_t)
+
+type qatlib_conf_t;
+files_config_file(qatlib_conf_t)
```
Comment (zpytela) [inline policy/modules/contrib/qatlib.te:13]: I can't see any rules for qatlib_t accessing qatlib_conf_t, is it correct?

File: policy/modules/contrib/qatlib.te [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+optional_policy(`
+        auth_read_passwd_file(qatlib_t)
+')
+
+optional_policy(`
+	miscfiles_read_localization(qatlib_t)
+')
+
+optional_policy(`
+	systemd_search_unit_dirs(qatlib_t)
```
Comment (zpytela) [inline policy/modules/contrib/qatlib.te:53]: Why does it need to search units directory? Is it because it wants to access some particular unit?

Comment (5umm3r15) [reply policy/modules/contrib/qatlib.te:53]: It is using awk command for checking systemd file.

AVC:
```
type=PATH msg=audit(1674238318.319:806): item=0 name="/usr/lib/systemd/system/qat.service" inode=159310 dev=fc:02 mode=0100644 ouid=0 ogid=0 rdev=00:00 obj=system_u:object_r:qatlib_unit_file_t:s0 nametype=NORMAL cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(1674238318.319:806): cwd="/"
type=SYSCALL msg=audit(1674238318.319:806): arch=c000003e syscall=257 success=yes exit=3 a0=ffffff9c a1=55dcf764e110 a2=0 a3=0 items=1 ppid=2592 pid=2593 auid=4294967295 uid=0 gid=992 euid=0 suid=0 fsuid=0 egid=992 sgid=992 fsgid=992 tty=(none) ses=4294967295 comm="awk" exe="/usr/bin/gawk" subj=system_u:system_r:qatlib_t:s0 key=(null)
type=AVC msg=audit(1674238318.319:806): avc:  denied  { search } for  pid=2593 comm="awk" name="system" dev="vda2" ino=137418 scontext=system_u:system_r:qatlib_t:s0 tcontext=system_u:object_r:systemd_unit_file_t:s0 tclass=dir permissive=1
```

File: policy/modules/contrib/qatlib.te (hunk 3 above)
Comment (5umm3r15) [reply policy/modules/contrib/qatlib.te:13]: I wasn't able to reproduce AVC messages related to config file

File: policy/modules/contrib/qatlib.fc (hunk 2 above)
Comment (5umm3r15) [reply policy/modules/contrib/qatlib.fc]: fixed, thank you

File: policy/modules/contrib/qatlib.fc (hunk 1 above)
Comment (5umm3r15) [reply policy/modules/contrib/qatlib.fc]: fixed, thank you

File: policy/modules/contrib/qatlib.te (hunk 3 above)
Comment (zpytela) [reply policy/modules/contrib/qatlib.te:13]: No problem with that, it just seems strange - why a service would not read its config?

**Comment by zpytela** [comment]
Merging, thank you.

**Pull request merged**